From 2e3c3ee164cd5eeaacb524ff0588650345ca3f66 Mon Sep 17 00:00:00 2001
From: Xianzhong <xianzhong.li@nxp.com>
Date: Sat, 8 Sep 2018 05:42:57 +0800
Subject: [PATCH 4599/5242] MGS-4151-9 [#imx-1166] merge kernel panic fix in
 GFP oom

commit  21b3c4c7dd0c1708fec91a553ec362692faedcb0 from
https://source.codeaurora.org/external/imx/linux-imx.git

CL167496 fix kernel panic caused by out-of-memory fix panic
 issue in bug #20884 comment #12

cherry-pick patch:
6.2.4.p3-0044-CL167496-fix-kernel-panic-caused-by-out-of-memory-fi

Signed-off-by: Xianzhong <xianzhong.li@nxp.com>
(cherry picked from commit f3a1c75a48e16441a939f20824fe1e2ebd5cc83d)
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 .../default/gc_hal_kernel_allocator_gfp.c          |    5 +++++
 1 file changed, 5 insertions(+)

diff --git a/drivers/mxc/gpu-viv/hal/os/linux/kernel/allocator/default/gc_hal_kernel_allocator_gfp.c b/drivers/mxc/gpu-viv/hal/os/linux/kernel/allocator/default/gc_hal_kernel_allocator_gfp.c
index a4fed0a..937c752 100644
--- a/drivers/mxc/gpu-viv/hal/os/linux/kernel/allocator/default/gc_hal_kernel_allocator_gfp.c
+++ b/drivers/mxc/gpu-viv/hal/os/linux/kernel/allocator/default/gc_hal_kernel_allocator_gfp.c
@@ -310,6 +310,11 @@ static int gc_usage_show(struct seq_file* m, void* data)
 
     mdlPriv = kzalloc(sizeof(struct gfp_mdl_priv), GFP_KERNEL | __GFP_NORETRY);
 
+    if (!mdlPriv)
+    {
+        gcmkONERROR(gcvSTATUS_OUT_OF_MEMORY);
+    }
+
 #if defined(CONFIG_ZONE_DMA32) && LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,37)
     if ((Flags & gcvALLOC_FLAG_4GB_ADDR) || (Allocator->os->device->platform->flagBits & gcvPLATFORM_FLAG_LIMIT_4G_ADDRESS))
     {
-- 
1.7.9.5

